cli/src/config.rs

```diff
@@ -217,6 +220,7 @@ impl LayeredConfigs {
             repo: None,
             env_overrides: env_overrides(),
             arg_overrides: None,
+            builtin_overrides: builtin_overrides(),
```

Comment:
Can't we just let user override the builtin if they really want? That's harmless anyway.

Even if we want to be strict about that, I (as a user) would expect some error/warning instead of silently shadowing the user-defined value.

Reply:
> Can't we just let user override the builtin if they really want? That's harmless anyway.

It feels like we'd end up in exactly the same place we started when #4162 was filed. No? Won't it be better to actually do nothing instead of that?

> 
> Even if we want to be strict about that, I (as a user) would expect some error/warning instead of silently shadowing the user-defined value.

I went this route. Could you take a look at where I hooked in the check? I feel like I just hit it on the head until it worked, but I don't know if there's a better place to insert this check? I also thought it may be a good idea to check for any user defined revset-aliases that start with `builtin_` instead of this particular alias.

Thoughts please! :-D

I still have to add tests and maybe some minor cleanup, but I want to check that there's no better way to do this first.

Thanks

Reply:
> > Can't we just let user override the builtin if they really want? That's harmless anyway.
> 
> It feels like we'd end up in exactly the same place we started when #4162 was filed. No? Won't it be better to actually do nothing instead of that?

It's not recommended to override builtins, but doing that wouldn't make the situation worse. That's my point. And silent failure is the worst thing IMHO because the user can't see why their configuration doesn't work.

> > Even if we want to be strict about that, I (as a user) would expect some error/warning instead of silently shadowing the user-defined value.
> 
> I went this route. Could you take a look at where I hooked in the check?

I think warnings can be emitted by `revset_util::load_revset_aliases()`.

BTW, if we implement this, `immutable()` and `mutable()` are eligible for warnings.

Reply:
> > > Can't we just let user override the builtin if they really want? That's harmless anyway.
> > 
> > 
> > It feels like we'd end up in exactly the same place we started when #4162 was filed. No? Won't it be better to actually do nothing instead of that?
> 
> It's not recommended to override builtins, but doing that wouldn't make the situation worse. That's my point. And silent failure is the worst thing IMHO because the user can't see why their configuration doesn't work.
> 
> > > Even if we want to be strict about that, I (as a user) would expect some error/warning instead of silently shadowing the user-defined value.
> > 
> > 
> > I went this route. Could you take a look at where I hooked in the check?
> 
> I think warnings can be emitted by `revset_util::load_revset_aliases()`.
> 
> BTW, if we implement this, `immutable()` and `mutable()` are eligible for warnings.

will a simple warning like: 'redefining `immutable()` and `mutable()` is not advised. redefine `immutable_heads()` instead' ?

Or does it make sense to also add `builtin_` versions of these as well?

```toml
'builtin_immutable()' = '::(immutable_heads() | root())'
'builtin_mutable()' = '~builtin_immutable()'
'immutable()' = 'builtin_immutable()'
'mutable' = 'builtin_mutable()'
```

Reply:
Done. PTAL.

Reply:
warnings added.